### PR TITLE
🚧Fix violations of RS1024 (Compare symbols correctly)

### PR DESCRIPTION
**Pull request by sharwell**
🚧 This is a _direct application_ of the Fix All operation from RS1024. It has not been modified in any way. Please review this with the intent of finding bugs in this new analyzer and/or code fix.

### CODE REVIEW

File: src/MetaCompilation.Analyzers/Core/DiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -2603,7 +2603,7 @@ private bool CheckMethods(InvocationExpressionSyntax invocationExpression, Compi
                         ReportDiagnostic(context, IncorrectAnalysisReturnTypeRule, analysisMethodSyntax.Identifier.GetLocation(), analysisMethodSyntax.Identifier.ValueText);
                         return false;
                     }
-                    else if (analysisMethod.Parameters.Count() != 1 || analysisMethod.Parameters.First().Type != context.Compilation.GetTypeByMetadataName("Microsoft.CodeAnalysis.Diagnostics.SyntaxNodeAnalysisContext"))
+                    else if (analysisMethod.Parameters.Count() != 1 || !Equals(analysisMethod.Parameters.First().Type, context.Compilation.GetTypeByMetadataName("Microsoft.CodeAnalysis.Diagnostics.SyntaxNodeAnalysisContext")))
```
Comment (mavasani) [inline src/MetaCompilation.Analyzers/Core/DiagnosticAnalyzer.cs]: I presume this analyzer is enabled by default (and also other analyzers you recently added)? If not, please ensure we turn them on for the repo by updating https://github.com/dotnet/roslyn-analyzers/blob/master/eng/Analyzers_ShippingRules.ruleset.

Please also update our Microsoft.CodeAnalysis.Analyzers [version](https://github.com/dotnet/roslyn-analyzers/blob/master/eng/Versions.props#L24) to the latest myget package so we enforce it on build. Thanks!

**Comment by mavasani** [comment]
retest this please

**Comment by mavasani** [comment]
Actually, there seem to be merge conflicts to resolve.

File: src/MetaCompilation.Analyzers/Core/DiagnosticAnalyzer.cs (hunk 1 above)
Comment (sharwell) [reply src/MetaCompilation.Analyzers/Core/DiagnosticAnalyzer.cs]: Yep, I'm working on that in a separate PR.

**Pull request closed without merging**